### PR TITLE
Refactor get_user to be more explicit about the upsert logic

### PR DESCRIPTION
**Pull request by marcospri**
Method local refactor, no test needs updating, no behaviour changes.

### CODE REVIEW

File: lms/services/user.py [hunk 1]
```diff
@@ -41,26 +41,23 @@ def upsert_user(self, lti_user: LTIUser) -> User:
...
 
-        user = User(
-            application_instance_id=lti_user.application_instance_id,
-            user_id=lti_user.user_id,
-            roles=lti_user.roles,
-            h_userid=lti_user.h_user.userid(self._h_authority),
-        )
-
-        if existing_user := self._db.execute(
+        user = self._db.execute(
```
Comment (marcospri) [inline lms/services/user.py:44]: Get the user

File: lms/services/user.py [hunk 2]
```diff
@@ -41,26 +41,23 @@ def upsert_user(self, lti_user: LTIUser) -> User:
...
             self._user_search_query(
-                application_instance_id=user.application_instance_id,
-                user_id=user.user_id,
+                application_instance_id=lti_user.application_instance_id,
+                user_id=lti_user.user_id,
+            )
+        ).scalar_one_or_none()
+
+        if not user:
```
Comment (marcospri) [inline lms/services/user.py:51]: If it doesn't exist, create a new object.

File: lms/services/user.py [hunk 3]
```diff
@@ -41,26 +41,23 @@ def upsert_user(self, lti_user: LTIUser) -> User:
...
             )
-        ).scalar_one_or_none():
-            # Update the existing user from the fields which can change on a
-            # new one
-            existing_user.roles = user.roles
-            user = existing_user
-        else:
             self._db.add(user)
 
+        user.roles = lti_user.roles
```
Comment (marcospri) [inline lms/services/user.py:60]: All the fields we'd update in all cases in the same block

File: lms/services/user.py [hunk 4]
```diff
@@ -41,26 +41,23 @@ def upsert_user(self, lti_user: LTIUser) -> User:
         # tied to users in groups. Should we start to store users who have not
         # launched us, we could inflate our numbers or change their meaning.
 
-        user = User(
```
Comment (marcospri) [inline lms/services/user.py:44]: Here we were constructing an new User object in all cases, I find that confusing.

**Review by seanh — APPROVED** [review]

**Pull request merged**